File: app/src/main/java/restaurant/sa/com/sarestaurant/appview/restaurant/model/Result.java

package restaurant.sa.com.sarestaurant.appview.restaurant.model;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Result {

    @Expose
    private Geometry geometry;
    @Expose
    private String icon;
    @Expose
    private String id;
    @Expose
    private String name;
    @Expose
    private List<Photo> photos;
    @SerializedName("place_id")
    private String placeId;
    @SerializedName("plus_code")
    private PlusCode plusCode;
    @SerializedName("opening_hours")
    @Expose
    private OpeningHours openingHours;
    @Expose
    private Double rating;
    @Expose
    private String reference;
    @Expose
    private String scope;
    @Expose
    private List<String> types;
    @Expose
    private String vicinity;
    public OpeningHours getOpeningHours() {
        return openingHours;
    }

    public void setOpeningHours(OpeningHours openingHours) {
        this.openingHours = openingHours;
    }


    public Geometry getGeometry() {
        return geometry;
    }

    public void setGeometry(Geometry geometry) {
        this.geometry = geometry;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Photo> getPhotos() {
        return photos;
    }

    public void setPhotos(List<Photo> photos) {
        this.photos = photos;
    }

    public String getPlaceId() {
        return placeId;
    }

    public void setPlaceId(String placeId) {
        this.placeId = placeId;
    }

    public PlusCode getPlusCode() {
        return plusCode;
    }

    public void setPlusCode(PlusCode plusCode) {
        this.plusCode = plusCode;
    }

    public Double getRating() {
        return rating;
    }

    public void setRating(Double rating) {
        this.rating = rating;
    }

    public String getReference() {
        return reference;
    }

    public void setReference(String reference) {
        this.reference = reference;
    }

    public String getScope() {
        return scope;
    }

    public void setScope(String scope) {
        this.scope = scope;
    }

    public List<String> getTypes() {
        return types;
    }

    public void setTypes(List<String> types) {
        this.types = types;
    }

    public String getVicinity() {
        return vicinity;
    }

    public void setVicinity(String vicinity) {
        this.vicinity = vicinity;
    }

}

File: app/src/main/java/restaurant/sa/com/sarestaurant/appview/restaurant/model/Result.kt
package restaurant.sa.com.sarestaurant.appview.restaurant.model

import android.arch.persistence.room.*
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

@Entity(tableName = "Result")
data class Result(
        @PrimaryKey(autoGenerate = true) var _id: Int,
        @Embedded var geometry: Geometry?,
        @ColumnInfo(name = "icon") var icon: String?,
        @ColumnInfo(name = "name") var name: String,
        @ColumnInfo(name = "id") var id: String?,
        @TypeConverters(PhotoConverters::class)
        var photos: List<Photo>?,
        @SerializedName("place_id")
        @Expose
        @ColumnInfo(name = "placeId") var placeId: String?,
        @Embedded var plusCode: PlusCode?,
        @Embedded var openingHours: OpeningHours?,
        @ColumnInfo(name = "rating") var rating: Double?,
        @ColumnInfo(name = "reference") var reference: String?,
        @ColumnInfo(name = "scope") var scope: String?,
        @TypeConverters(TypesConverters::class)
        var types: List<String>?,
        @ColumnInfo(name = "vicinity") var vicinity: String?)
